packages/core/test/waitFor.test.ts

```diff
@@ -57,30 +54,22 @@ describe('waitFor', () => {
           on: { NEXT: 'b' }
         },
         b: {
-          on: { NEXT: 'c' }
-        },
-        c: {
           type: 'final'
         }
       }
     });
 
     const service = interpret(machine).start();
 
-    setInterval(() => {
+    setTimeout(() => {
       service.send('NEXT');
     }, 10);
 
-    setTimeout(() => {
-      throw new Error('Should not reach here');
-    }, 50);
```

Comment:
similarly, this was never disposed and could be thrown when **other** test was executing and thus fail that other test. I don't think this was adding that much here and by using `await expect().rejects()` we make sure that this `waitFor` throws